### PR TITLE
Provide more ways to set AWS credentials in Functionbeat

### LINKED ISSUE #12464: Please support AWS IAM Instance profiles!
I can't use the Functionbeat on my EC2 server because it doesn't appear to use the default credential provider chain.  Meaning that I have a Role assigned to me EC2 instance that gives it permission to access the bucket specified in the configuration, but the deploy function doesn't see use it and therefore has no permissions to do anything with S3.

I'm trying to deploy the functionbeat from our EC2 server and it's not possible given this missing functionality.  I've got my entire ELK stack deployment scripted except for this functionbeat.

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR makes credential settings when deploying Lambdas to AWS more flexible. New options are introduced:

1. `access_key_id`, `secret_access_key` and/or `session_token` for tokens

```yaml
functionbeat.provider.aws.access_key_id: '${AWS_ACCESS_KEY_ID:""}'
functionbeat.provider.aws.secret_access_key: '${AWS_SECRET_ACCESS_KEY:""}'
functionbeat.provider.aws.session_token: '${AWS_SESSION_TOKEN:""}'
``` 
2. `role_arn` for assuming IAM roles
```yaml
functionbeat.provider.aws.role_arn: arn:aws:iam::123456789012:role/test-fnb
```

3. `credential_profile_name` and/or `shared_credential_file` for credential files
```yaml
functionbeat.provider.aws.credential_profile_name: fnb-aws
functionbeat.provider.aws.shared_credential_file: /etc/functionbeat/aws_credentials
```

## Why is it important?

Credential configuration becomes more flexible and follows the same pattern as in Filebeat and Metricbeat.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

Based on #17658
Closes #12464

### CODE REVIEW

**Comment by kvch** [comment]
Some tests need adjustment.

File: x-pack/functionbeat/provider/aws/aws/config.go [hunk 1]
```diff
@@ -15,12 +15,13 @@ import (
...
 	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	awscommon "github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
 )
 
 // Config expose the configuration option the AWS provider.
 type Config struct {
-	Endpoint     string `config:"endpoint" validate:"nonzero,required"`
-	DeployBucket bucket `config:"deploy_bucket" validate:"nonzero,required"`
+	DeployBucket bucket              `config:"deploy_bucket" validate:"nonzero,required"`
+	Credentials  awscommon.ConfigAWS `config:",inline"`
```
Comment (kaiyan-sheng) [inline x-pack/functionbeat/provider/aws/aws/config.go:24]: Credential part looks good but I'm wondering if we should add a check for endpoint part. Endpoint is not required in `awscommon.ConfigAWS` but seems like should be required here.

**Review by kaiyan-sheng — APPROVED** [review]
Looks good to me, not sure why doc is not happy though:
```
12:32:16 INFO:build_docs:asciidoctor: ERROR: config-options-aws.asciidoc: line 219: include file not found: /tmp/docsbuild/eSoev3Xy72/beats/x-pack/libbeat/docs/aws-credentials-config.asciidoc
```

**Comment by bmorelli25** [comment]
Wow, this was a fun doc build error to try and hunt down. I _think_ you're seeing this issue because no file named `aws-credentials-examples.asciidoc` exists at `/functionbeat/docs`. 

You're including `aws-credentials-config.asciidoc`: `/beats/x-pack/libbeat/docs/aws-credentials-config.asciidoc` which has this include on line 46: `include::../../../{beatname_lc}/docs/aws-credentials-examples.asciidoc[]`

**Comment by bmorelli25** [comment]
Pushed a partial fix in dd080ba. From here, we'll merge https://github.com/elastic/docs/pull/2035, run a doc build, and finally rerun the preview build on this PR.

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
New error:
```
14:46:32 INFO:build_docs:asciidoctor: ERROR: ../../libbeat/docs/aws-credentials-config.asciidoc: line 59: include file not found: /tmp/docsbuild/rCldsYRINg/beats/filebeat/docs/aws-credentials-examples.asciidoc
```

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Pull request merged**